yt/visualization/plot_modifications.py

```diff
@@ -1806,11 +1806,12 @@ def __call__(self, plot):
         xx0, xx1, yy0, yy1 = self._plot_bounds(plot)
         # normalize all of the kwarg lengths to the plot size
         plot_diag = ((yy1 - yy0) ** 2 + (xx1 - xx0) ** 2) ** (0.5)
-        self.length *= plot_diag
-        self.width *= plot_diag
-        self.head_width *= plot_diag
-        if self.head_length is not None:
-            self.head_length *= plot_diag
+        length = self.length * plot_diag
+        width = self.width * plot_diag
+        head_width = self.head_width * plot_diag
+        head_length = None
+        if self.head_length:
```

Comment:
I think using truthiness here is a breaking change because it's a change of behaviour when passing `head_length=0`
```suggestion
        if self.head_length is not None:
```